src/dcqc/file.py

```diff
@@ -308,6 +309,27 @@ def is_file_local(self) -> bool:
         """
         return self._local_path is not None
 
+    def already_staged(self) -> list[Path]:
+        """Check if the target file has already been staged to the remote directory.
+
+        Returns:
+            staged_file_paths (list): List of already staged file paths.
+            Empty list if file has not been staged.
+
+        Raises:
+            FileExistsError: If the file has already been staged more than once.
+            This would cause a name collision in Nextflow.
+        """
+        path_str = os.path.join(gettempdir(), self.tmp_dir + "*", self.name)
```

Comment:
Is the star a risk for files that have the same name but different content?

Reply:
the star subs for the random string that is appended to the `dcqc-staged-` prefix so the path is then `dcqc-staged-*/file.txt`. So it should only get those paths where the same file has been staged in a different parent directory.